### ISSUE TITLE
App Store Rejection

### ISSUE THREAD

**Issue by alanwjlu** [LonelyCpp/react-native-youtube-iframe#127]
My app uses react-native-youtube-iframe to play Youtube videos in it. An update to my App got rejected today. Here's the message from Apple:

> Guideline 5.2.3 - Legal
> 
> 
> Your app contains content or features that may violate the rights of one or more third parties. Specifically, your app provides potentially unauthorized access to third-party audio or video streaming, catalogs, and discovery services.
> 
> Your app and its contents should not infringe upon the rights of another party. In the event your app infringes another party’s rights, you are responsible for any liability to Apple because of a claim.
> 
> Next Steps
> 
> To resolve this issue, please attach documentary evidence in the App Review Information section in App Store Connect evidencing that you have all necessary rights or permissions to the third-party audio or video streaming, catalogs, and discovery services in your app.

Has anyone faced this issue while using react-native-youtube-iframe? Any solutions / suggestions? Also found this thread where people were discussing similar problem, although not with react native: https://github.com/0xced/XCDYouTubeKit/issues/485

**Comment by LonelyCpp** [comment]
We've had a couple of android rejections (there's a section in the documentation for this)

But an app store rejection is a first.

 make sure that 

* you do not have any kind of "download" or "play in the background" feature in the app.
* no offline cache features
* only uses the webview iframe from this library. (as it's the recommended way on iOS)

I believe you'll have have to show that the app does nothing to break youtube's ToS to get approved. Please keep us posted

**Comment by aakashr143** [comment]
I have a app on the App Store which steams videos off YouTube

I think you got reject because you didn't have the right permission, for example if you are playing videos from a channel on YouTube, you should give apple a document saying that the channel doesnt have a problem with you using their videos in your app. 

What I did was, I had the channel upload a video (private) on their channel saying that they don't have a problem with me using their videos, add your Apple ID on screen (just in case), and I gave that link to apple.

**Comment by alanwjlu** [comment]
@aakashr143 in your case did you know the owner of the channel that you are streaming to your app so you can ask them to do this for you? And did you have to do that for every video / channels that you show in your app?

**Comment by LonelyCpp** [comment]
I found this youtube compliance audit - https://support.google.com/youtube/contact/yt_api_form?hl=en

I believe you can fill this out and get an approval from Google. You can then attach it while appealing the app store review.

**Comment by LonelyCpp** [comment]
is playing youtube videos a core feature of your app?

**Comment by aakashr143** [comment]
Yes, I knew the owner of the channel, and you will need the permission once (even if there are multiple channels by the person), Would it be possible for you to tell what your app does

**Comment by ShimTeacher** [comment]
Is there a developer who has passed the screening using this library?

**Comment by LonelyCpp** [comment]
@ShimTeacher 

I have production apps on the app store without any problems. But youtube is not a central part of the apps, it's mainly used for explainers and product related videos from the channel that owns the app.

did your app get rejected too!?

**Comment by ShimTeacher** [comment]
@LonelyCpp No, I haven't distributed it yet, and I asked a question while I was looking into the YouTube library.

**Comment by denkeni** [comment]
I've confirmed that App Store Review has revised their internal policy to approve 3rd party apps for YouTube streaming, just give them [YouTube API Services Terms of Service](https://developers.google.com/youtube/terms/api-services-terms-of-service). More details here: https://github.com/youtube/youtube-ios-player-helper/issues/399#issuecomment-867539604

**Comment by rafeeJ** [comment]
Quick questions regarding picture in picture for the player. I've read the documentation regarding the necessary steps to avoid the video playing in the background, and implemented this. 

However, I noticed that I can select picture in picture mode for the YouTube video I am playing from within the app. Would this be an issue raised during the review process?

**Comment by TSB1999** [comment]
> Quick questions regarding picture in picture for the player. I've read the documentation regarding the necessary steps to avoid the video playing in the background, and implemented this. 
> 
> 
> 
> However, I noticed that I can select picture in picture mode for the YouTube video I am playing from within the app. Would this be an issue raised during the review process?

Hey there. How did you manage to get Picture in Picture working with this library?

**Comment by deckameron** [comment]
> I've confirmed that App Store Review has revised their internal policy to approve 3rd party apps for YouTube streaming, just give them [YouTube API Services Terms of Service](https://developers.google.com/youtube/terms/api-services-terms-of-service). More details here: [youtube/youtube-ios-player-helper#399 (comment)](https://github.com/youtube/youtube-ios-player-helper/issues/399#issuecomment-867539604)

Hi @denkeni,

I did exactly what you said, and even submitting the [YouTube API Services Terms of Service](https://developers.google.com/youtube/terms/api-services-terms-of-service), they rejected stating the same thing. It is like a broken record. Apparently the issue is back!

Have you had any issues in the past couple of years? If so, what did you do to solve it?
Thank you!

**Comment by denkeni** [comment]
Try [appealing](https://developer.apple.com/distribute/app-review/#appeals). App Store Reviewers are unpredictable.
